04.bowling/bowling.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+def main
+  scores = ARGV[0].split(',')
+
+  shots = replace_strike_to_score(scores)
+  frames = slice_each_frame(shots)
+  p calculate_total_score(frames)
+end
+
+def replace_strike_to_score(scores)
+  shots = []
+
+  scores.each do |score|
+    if score == 'X'
+      shots << 10
+      shots << 0 if shots.size.odd?
+    else
+      shots << score.to_i
+    end
+  end
+
+  shots
+end
+
+def slice_each_frame(shots)
+  frames = []
+
+  shots.each_slice(2) do |shot|
+    shot.pop if shot[0] == 10 && shot[1].zero?
+    frames << shot
+  end
+
+  until frames.size == 10
+    frames[9].concat(frames[10])
+    frames.delete_at(10)
+  end
+
+  frames
+end
+
+def calculate_total_score(frames)
+  total = 0
+
+  # 次のフレームや、その次のフレームを参照しやすくするためにeachではなくuntil emptyで回しています
+  until frames.empty?
+    frame = frames.shift
+    total += frames.size.zero? ? frame.sum : calculate_each_frame(frame, frames)
```

Comment:
```suggestion
    total += calculate_each_frame(frame, frames)
```

IMOなコメントですが
せっかく `frame` と `frames` を受け取ってスコアを計算するメソッドを用意しているので
三項演算子で挙動を切り替えない方がわかりやすいかな？と思いました。

とはいえこのままでもOKです！

Reply:
10フレーム目かどうかの判定のためにこの三項演算子を使用しています！自分で読んでも分かりにくかったので、if文にしたりコメントを書くべきでした！💦